exercises/dnd-character/src/test/java/DnDCharacterTest.java

```diff
@@ -128,7 +133,224 @@ public void testRandomCharacterIsValid() {
     @Ignore("Remove to run test")
     @Test
     public void testEachAbilityIsOnlyCalculatedOnce() {
-        assertEquals(dndCharacter.getStrength(), dndCharacter.getStrength());
+        int initialStrength     = dndCharacter.getStrength();
+        int initialDexterity    = dndCharacter.getDexterity();
+        int initialConstitution = dndCharacter.getConstitution();
+        int initialIntelligence = dndCharacter.getIntelligence();
+        int initialWisdom       = dndCharacter.getWisdom();
+        int initialCharisma     = dndCharacter.getCharisma();
+        for (int j = 0; j < 100; j++) {
+            assertEquals(initialStrength,     dndCharacter.getStrength());
+            assertEquals(initialDexterity,    dndCharacter.getDexterity());
+            assertEquals(initialConstitution, dndCharacter.getConstitution());
+            assertEquals(initialIntelligence, dndCharacter.getIntelligence());
+            assertEquals(initialWisdom,       dndCharacter.getWisdom());
+            assertEquals(initialCharisma,     dndCharacter.getCharisma());
+        }
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testAbilityIsProbablyNotConstant() {
+        int ability = dndCharacter.ability();
+        for (int i = 0; i < 100; i++) {
+            if (ability != dndCharacter.ability()) {
+                return;
+            }
+        }
+        fail();
```

Comment:
This should probably include an error message saying what happened so that the student could debug the test failure: "ability appears to always return the same value"